### PR TITLE
Add fdm (fetch deliver mail) and dependency

### PR DESCRIPTION
**Pull request by gabeguz**
fdm is a small program for fetching mail from remote stores (imap and pop3) and delivering them to local stores (maildir, mbox, etc).  It's a very nice way to get your mail from mulitple sources and deposit it into your local mail store w/a bit of processing as well.  Especially for us CLI users. 

Let me know if there are any changes you need me to make in order to accept this.

### CODE REVIEW

File: Library/Formula/tdb.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                          "--host=6x86"
+    system "make", "install"
+  end
+
+  # older versions of the compiler allowed you to have a printf that
+  # spaned multiple lines, now that's no longer the case. There is a
+  # patch submitted upstream for this but it doesn't seem like it will
+  # ever be applied: http://sourceforge.net/p/tdb/patches/6/ as it
+  # dates back to 2003, and still hasn't been.
+  patch :DATA
```
Comment (DomT4) [inline Library/Formula/tdb.rb]: ` patch :DATA ` has to go above ` def install `.

**Comment by gabeguz** [comment]
Thanks @DomT4 -- I even read about that, and still messed it up!  
Fixed.

File: Library/Formula/fdm.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class Fdm < Formula
+  homepage "http://sourceforge.net/projects/fdm/"
+  url "https://downloads.sourceforge.net/project/fdm/fdm/fdm-1.8/fdm-1.8.tar.gz"
+  sha1 "25c0ebc0c2b43984d04c82d0ba3909acf7403df7"
+
+  depends_on "tdb"
```
Comment (mikemcquaid) [inline Library/Formula/fdm.rb]: Is `tdb` used by much/any software other than `fdm`? If not, we should probably install it as a `resource` of the `fdm` formula.

File: Library/Formula/fdm.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    # fdm 1.8 uses an environment variable to set $PREFIX,
+    # the --prefix param to configure is ignored.
+    ENV["PREFIX"] = "#{prefix}"
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules"
+    system "make"
+    bin.install "fdm"
+    man1.install "fdm.1"
+    man5.install "fdm.conf.5"
```
Comment (mikemcquaid) [inline Library/Formula/fdm.rb]: Is there no `make install`?

File: Library/Formula/tdb.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--mandir=#{man}",
+                          "--host=6x86"
```
Comment (mikemcquaid) [inline Library/Formula/tdb.rb]: Is this hardcoding the arch to `i686`? I don't think we want that.

**Comment by mikemcquaid** [comment]
I'm hesistant to include something where upstream seems to be dead and it requires a patch to run.

Also, can `test do` blocks be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/fdm.rb (hunk 3 above)
Comment (gabeguz) [inline Library/Formula/fdm.rb]: @mikemcquaid There is a make install, but it puts the man pages in man/ and not in share/man and there isn't a configure option to support man path changes, so I had to copy things explicitly.

File: Library/Formula/fdm.rb (hunk 2 above)
Comment (gabeguz) [inline Library/Formula/fdm.rb]: @mikemcquaid Hmm... it seems like tdb was originally created as part of the samba project... it's possible that samba has a more up to date implementation.  I'll dig around a bit more and see what I can find.

File: Library/Formula/tdb.rb (hunk 4 above)
Comment (gabeguz) [inline Library/Formula/tdb.rb]: Was the only way I could get the configure script to not error out.  The auto host detection doesn't understand x86_64 -- I'll see if there's a better way I can deal with this.  Thanks @mikemcquaid

File: Library/Formula/fdm.rb (hunk 3 above)
Comment (mikemcquaid) [inline Library/Formula/fdm.rb]: Let's use their `make install` and use `inreplace` or a patch to fix up their Makefile and submit these changes upstream.

**Comment by gabeguz** [comment]
So, I've gotten stuck because for some reason, fdm is being linked improperly to tdb, I can brew install tdb w/out a problem and brew install fdm w/out a problem, but when I try to run fdm, I get this: 

```
dyld: Library not loaded: /usr/local/opt/tdb/lib/libtdb.inst.dylib
Referenced from: /usr/local/bin/fdm
Reason: image not found
Trace/BPT trap: 5
```

And indeed, `/usr/local/opt/tdb/lib/libtdb.inst.dylib` does not exist: 

```
$ ls /usr/local/opt/tdb/lib/
libtdb.1.3.4.dylib      libtdb.1.dylib          libtdb.dylib            pkgconfig               python2.7
```

If I use `install_name_tool` to change the path of the library: 
```
# install_name_tool -change /usr/local/opt/tdb/lib/libtdb.inst.dylib /usr/local/opt/tdb/lib/libtdb.dylib ./fdm
```

Then fdm works fine.  
I'm guessing that I am doing something wrong when I'm building fdm, but I don't know what... any pointers?

**Comment by bfontaine** [comment]
Any update on this?

**Comment by gabeguz** [comment]
@bfontaine I'm still stuck on the issue I was seeing... I'll give it another go and see if I can figure it out.

**Comment by DomT4** [comment]
Update here?

**Comment by cvogt** [comment]
looks like more recent versions of fdm van be found here: https://github.com/nicm/fdm

**Comment by dunn** [comment]
Is that the new canonical location?

**Comment by DomT4** [comment]
Looks like it [may well be](https://sourceforge.net/p/fdm/mailman/message/34179566/).

**Comment by DomT4** [comment]
Will review a fresh PR on this if anyone is still interested. Actually looks like there's a new release now they've moved to Github. A lot has changed since these formulae were submitted originally so you'd need to rebase against an up-to-date master & then `brew audit --strict --online` them.

Thanks for the submission! :heart:

**Pull request closed without merging**